internal/code-generator/generator/heuristics_test.go

```diff
@@ -0,0 +1,129 @@
+package generator
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"testing"
+	"text/template"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestYamlParsing(t *testing.T) {
+	heuristics, err := parseYaml("test_files/heuristics.yml")
+
+	require.NoError(t, err)
+	assert.NotNil(t, heuristics)
+
+	// extensions
+	require.NotNil(t, heuristics.Disambiguations)
+	assert.Equal(t, 4, len(heuristics.Disambiguations))
+	assert.Equal(t, 2, len(heuristics.Disambiguations[0].Extensions))
+
+	rules := heuristics.Disambiguations[0].Rules
```

Comment:
Please add a comment to the heuristics.yml file to document that the test cares about the number and order of entries.